tsdb/testutil.go

```diff
@@ -0,0 +1,195 @@
+	sampleType string
+	appendFunc func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error)
+	sampleFunc func(ts, value int64) sample
+}
+
+var sampleTypeScenarios = map[string]sampleTypeScenario{
+	float: {
+		sampleType: sampleMetricTypeFloat,
+		appendFunc: func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error) {
+			s := sample{t: ts, f: float64(value)}
+			ref, err := appender.Append(0, lbls, ts, s.f)
+			return ref, s, err
+		},
+		sampleFunc: func(ts, value int64) sample {
+			return sample{t: ts, f: float64(value)}
+		},
+	},
+	// intHistogram: {
+	//	 sampleType: sampleMetricTypeHistogram,
+	//	 appendFunc: func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error) {
+	//		 s := sample{t: ts, h: tsdbutil.GenerateTestHistogram(int(value))}
+	//		 ref, err := appender.AppendHistogram(0, lbls, ts, s.h, nil)
+	//		 return ref, s, err
+	//	 },
+	//	 sampleFunc: func(ts, value int64) sample {
+	//		 return sample{t: ts, h: tsdbutil.GenerateTestHistogram(int(value))}
+	//	 },
+	// },
+	// floatHistogram: {
+	//	 sampleType: sampleMetricTypeHistogram,
+	//	 appendFunc: func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error) {
+	//		 s := sample{t: ts, fh: tsdbutil.GenerateTestFloatHistogram(int(value))}
+	//		 ref, err := appender.AppendHistogram(0, lbls, ts, nil, s.fh)
+	//		 return ref, s, err
+	//	 },
+	//	 sampleFunc: func(ts, value int64) sample {
+	//		 return sample{t: ts, fh: tsdbutil.GenerateTestFloatHistogram(int(value))}
+	//	 },
+	// },
+	// gaugeIntHistogram: {
+	//	 sampleType: sampleMetricTypeHistogram,
+	//	 appendFunc: func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error) {
+	//		 s := sample{t: ts, h: tsdbutil.GenerateTestGaugeHistogram(int(value))}
+	//		 ref, err := appender.AppendHistogram(0, lbls, ts, s.h, nil)
+	//		 return ref, s, err
+	//	 },
+	//	 sampleFunc: func(ts, value int64) sample {
+	//		 return sample{t: ts, h: tsdbutil.GenerateTestGaugeHistogram(int(value))}
+	//	 },
+	// },
+	// gaugeFloatHistogram: {
+	//	 sampleType: sampleMetricTypeHistogram,
+	//	 appendFunc: func(appender storage.Appender, lbls labels.Labels, ts, value int64) (storage.SeriesRef, sample, error) {
+	//		 s := sample{t: ts, fh: tsdbutil.GenerateTestGaugeFloatHistogram(int(value))}
+	//		 ref, err := appender.AppendHistogram(0, lbls, ts, nil, s.fh)
+	//		 return ref, s, err
+	//	 },
+	//	 sampleFunc: func(ts, value int64) sample {
+	//		 return sample{t: ts, fh: tsdbutil.GenerateTestGaugeFloatHistogram(int(value))}
+	//	 },
+	// },
+}
+
+// requireEqualSamples checks that the actual series are equal to the expected ones. It ignores the counter reset hints for histograms.
+func requireEqualSamples(t *testing.T, expected, actual map[string][]chunks.Sample, ignoreCounterResets bool) {
```

Comment:
Maybe this should be renamed to `requireEqualSeries` because you have to pass in the series name as well as the samples. And then `compareSamples` can be renamed to `requireEqualSamples`.